sfnt/table_feature.go

```diff
@@ -0,0 +1,417 @@
+package sfnt
+
+import (
+	"bytes"
+	"encoding/binary"
+	"fmt"
+	"io"
+	"io/ioutil"
+)
+
+// TableLayout represents the common layout table used by GPOS and GSUB.
+// The Features field contains all the features for this layout. However,
+// the script and language determines which feature is used.
+//
+// See https://www.microsoft.com/typography/otspec/chapter2.htm#organization
+// See https://www.microsoft.com/typography/otspec/GPOS.htm
+// See https://www.microsoft.com/typography/otspec/GSUB.htm
+type TableLayout struct {
+	bytes   []byte
+	version versionHeader
+	header  layoutHeader11
+
+	Scripts  []*Script  // Scripts contains all the scripts in this layout.
+	Features []*Feature // Features contains all the features in this layout.
+	Lookups  []*Lookup  // Lookups contains all the lookups in this layout.
+}
+
+// Bytes returns the bytes for this table. The TableLayout is read only, so
+// the bytes will always be the same as what is read in.
+func (t *TableLayout) Bytes() []byte {
+	return t.bytes
+}
+
+// Script represents a single script (i.e "latn" (Latin), "cyrl" (Cyrillic), etc).
+type Script struct {
+	Tag             Tag        // Tag for this script.
+	DefaultLanguage *LangSys   // DefaultLanguage used by this script.
+	Languages       []*LangSys // Languages within this script.
+}
+
+// Script returns the name for this script.
+func (s *Script) String() string {
+	return scriptTags[s.Tag.String()]
+}
+
+// LangSys represents the language system for a script.
+type LangSys struct {
+	Tag      Tag        // Tag for this language.
+	Features []*Feature // Features contains the features for this language.
+}
+
+// String returns the name for this language.
+func (l *LangSys) String() string {
+	return languageTags[l.Tag.String()]
+}
+
+// Feature represents a glyph substitution or glyph positioning features.
+type Feature struct {
+	Tag Tag // Tag for this feature
+}
+
+// Script returns the name for this feature.
+func (f *Feature) String() string {
+	tag := f.Tag.String()
+	if name, found := featureTags[tag]; found {
+		return name
+	}
+
+	if len(tag) == 4 && tag[0] == 'c' && tag[1] == 'v' && tag[2] >= '0' && tag[2] <= '9' && tag[3] >= '0' && tag[3] <= '9' {
+		return "Character Variants"
```

Comment:
@bramp might be worth preserving the number here (e.g. Character Variables 01)

Reply:
(Also, should we do the same parsing for `ss01`...? )

Reply:
Sure, will fix.

Reply:
Added numeric suffix, plus added support for ss01...